### PR TITLE
Fix incorrect types and add currency pairs

### PR DESCRIPTION
**Pull request by m4ce**
As per https://docs.bitbank.cc, I am adding a number of missing instruments. As I am relying on your Java API to interact with your backend, it'd be great if you could make a release out of it..

### CODE REVIEW

**Comment by m4ce** [comment]
I've also fixed the Transaction class where price and amounts are actually strings..

File: src/main/java/cc/bitbank/entity/Depth.java [hunk 1]
```diff
@@ -12,9 +12,9 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class Depth extends Data {
 
-    public BigDecimal[][] asks;
-    public BigDecimal[][] bids;
-    public Date timestamp;
+    private BigDecimal[][] asks;
```
Comment (junderw) [inline src/main/java/cc/bitbank/entity/Depth.java:15]: Please refrain from changing the interfaces if not necessary.

File: src/main/java/cc/bitbank/entity/Transactions.java [hunk 2]
```diff
@@ -15,21 +15,41 @@
...
     public static class Transaction {
         @JsonProperty("transaction_id")
-        public long transactionId;
+        private long transactionId;
 
-        public OrderSide side;
+        private OrderSide side;
 
-        public int price;
+        private String price;
```
Comment (junderw) [inline src/main/java/cc/bitbank/entity/Transactions.java:22]: Keep it public.
Change to String is ok.

File: src/main/java/cc/bitbank/entity/Transactions.java [hunk 3]
```diff
@@ -15,21 +15,41 @@
...
 
-        public BigDecimal amount;
+        private String amount;
```
Comment (junderw) [inline src/main/java/cc/bitbank/entity/Transactions.java:24]: Keep it public.
Change to String is ok.

File: src/main/java/cc/bitbank/entity/Transactions.java [hunk 4]
```diff
@@ -38,4 +58,8 @@ public Transactions () {}
     public Transactions (Transaction[] transactions) {
         this.transactions = transactions;
     }
+
+    public Transaction[] getTransactions() {
```
Comment (junderw) [inline src/main/java/cc/bitbank/entity/Transactions.java:62]: Please remove along with all the other new methods. (this should be a separate pull request)

File: src/main/java/cc/bitbank/entity/enums/CurrencyPair.java [hunk 5]
```diff
@@ -7,7 +7,12 @@
 public enum CurrencyPair {
     BTC_JPY("btc_jpy"),
     LTC_BTC("ltc_btc"),
-    XRP_JPY("xrp_jpy");
+    XRP_JPY("xrp_jpy"),
```
Comment (junderw) [inline src/main/java/cc/bitbank/entity/enums/CurrencyPair.java:10]: Thanks for this!

**Review by junderw — CHANGES_REQUESTED** [review]
1. Adding CurrencyPairs
2. Changing from int and BigDecimal to String x 2

These are ok. Everything else, please separate into a new pull request.

In general, "Minor Improvements" with a lot of changes are less likely to get merged. Each pull request should have an explicit purpose.

Thanks.

**Review by junderw — APPROVED** [review]
nvm got the OK to just approve everything.

**Comment by m4ce** [comment]
Thank you! We have other changes coming up. We will make a separate PR for these.

**Pull request merged**